include/margo-timer.h

```diff
@@ -50,7 +78,12 @@ int margo_timer_create(margo_instance_id       mid,
 int margo_timer_start(margo_timer_t timer, double timeout_ms);
 
 /**
- * @brief Cancel a started timer.
+ * @brief Cancel a started timer. If the timer's callback
+ * has already been submitted as a ULT, this ULT will
+ * eventually be executed. Hence, calling margo_timer_cancel
```

Comment:
I think this function comment is incorrect.  There is no wait_pending() function.  By my reading of the code it will block until timer ULTs are complete (if they have been launched already).  I like that because it is much simpler anyway.

In place of the note about wait_pending() it should probably warn callers that this particular could block (in an Argobots-safe way, that is).

Reply:
Good catch! I forgot to update this comment.